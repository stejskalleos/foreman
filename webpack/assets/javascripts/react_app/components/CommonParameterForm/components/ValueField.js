import React, { useState } from 'react';
import { Modal } from 'patternfly-react';
import PropTypes from 'prop-types';

import { translate as __ } from '../../../common/I18n';
import FormField from '../../common/forms/FormField';

import {
  fieldSizeByType,
  valueFieldByType,
  showFullScreenBtn,
} from '../CommonParamFormHelpers';

const ValueField = ({ value, selectedType, isMasked, onChange, error }) => {
  const [showModal, setShowModal] = useState(false);
  const valueFieldProps = {
    id: 'common_parameter_value',
    name: 'common_parameter[value]',
    css: 'form-control',
    value,
    onChange,
    isMasked,
  };

  return (
    <React.Fragment>
      <FormField
        label={__('Value')}
        inputSizeClass={fieldSizeByType(selectedType)}
        error={error}
      >
        {valueFieldByType(selectedType, valueFieldProps)}
        {showFullScreenBtn(selectedType, setShowModal)}
      </FormField>
      <Modal
        show={showModal}
        onHide={() => setShowModal(false)}
        className="editor-modal"
      >
        <Modal.Header closeButton />
        <Modal.Body>
          {valueFieldByType(selectedType, {
            ...valueFieldProps,
            fullScreen: true,
          })}
        </Modal.Body>
      </Modal>
    </React.Fragment>
  );
};

ValueField.propTypes = {
  value: PropTypes.any,
  selectedType: PropTypes.string,
  isMasked: PropTypes.bool,
  onChange: PropTypes.func.isRequired,
  error: PropTypes.string,
};

ValueField.defaultProps = {
  value: '',
  selectedType: '',
  isMasked: false,
  error: '',
};

export default ValueField;
