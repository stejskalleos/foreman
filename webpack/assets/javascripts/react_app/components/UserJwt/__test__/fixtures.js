import { noop } from '../../../common/helpers';
import { STATUS } from '../../../constants';

export const formProps = {
  handleSubmit: noop,
};

const token = 'token23';

export const fieldOk = {
  apiStatus: STATUS.RESOLVED,
  token,
};

export const fieldError = {
  apiStatus: STATUS.ERROR,
  token,
};

export const fieldPending = {
  apiStatus: STATUS.PENDING,
  token,
};
