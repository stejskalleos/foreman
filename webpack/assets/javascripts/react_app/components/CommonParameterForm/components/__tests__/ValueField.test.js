import { testComponentSnapshotsWithFixtures } from '@theforeman/test';

import { valueFieldFixtures } from './fixtures';
import ValueField from '../ValueField';

describe('ValueField', () => {
  describe('rendering', () =>
    testComponentSnapshotsWithFixtures(ValueField, valueFieldFixtures));
});
