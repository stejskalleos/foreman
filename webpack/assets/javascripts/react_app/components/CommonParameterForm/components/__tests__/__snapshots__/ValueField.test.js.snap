// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ValueField rendering default 1`] = `
<Fragment>
  <FormField
    className=""
    disabled={false}
    error=""
    helpInline={null}
    id={null}
    inputProps={null}
    inputSizeClass="col-md-8"
    label="Value"
    labelHelp={null}
    labelSizeClass="col-md-2"
    onChange={[Function]}
    required={false}
    type="text"
  >
    <StringValue
      css="form-control"
      fullScreen={false}
      id="common_parameter_value"
      isMasked={false}
      name="common_parameter[value]"
      onChange={[Function]}
      value=""
    />
    <Button
      active={false}
      block={false}
      bsClass="btn"
      bsStyle="default"
      disabled={false}
      onClick={[Function]}
    >
      <Icon
        name="expand"
        type="fa"
      />
       Full screen
    </Button>
  </FormField>
  <Modal
    animation={true}
    autoFocus={true}
    backdrop={true}
    bsClass="modal"
    className="editor-modal"
    dialogComponentClass={[Function]}
    enforceFocus={true}
    keyboard={true}
    manager={
      ModalManager {
        "add": [Function],
        "containers": Array [],
        "data": Array [],
        "handleContainerOverflow": true,
        "hideSiblingNodes": true,
        "isTopModal": [Function],
        "modals": Array [],
        "remove": [Function],
      }
    }
    onHide={[Function]}
    renderBackdrop={[Function]}
    restoreFocus={true}
    show={false}
  >
    <ModalHeader
      bsClass="modal-header"
      closeButton={true}
      closeLabel="Close"
    />
    <ModalBody
      bsClass="modal-body"
      componentClass="div"
    >
      <StringValue
        css="form-control"
        fullScreen={true}
        id="common_parameter_value"
        isMasked={false}
        name="common_parameter[value]"
        onChange={[Function]}
        value=""
      />
    </ModalBody>
  </Modal>
</Fragment>
`;

exports[`ValueField rendering when is hidden 1`] = `
<Fragment>
  <FormField
    className=""
    disabled={false}
    error=""
    helpInline={null}
    id={null}
    inputProps={null}
    inputSizeClass="col-md-8"
    label="Value"
    labelHelp={null}
    labelSizeClass="col-md-2"
    onChange={[Function]}
    required={false}
    type="text"
  >
    <StringValue
      css="form-control"
      fullScreen={false}
      id="common_parameter_value"
      isMasked={true}
      name="common_parameter[value]"
      onChange={[Function]}
      value=""
    />
    <Button
      active={false}
      block={false}
      bsClass="btn"
      bsStyle="default"
      disabled={false}
      onClick={[Function]}
    >
      <Icon
        name="expand"
        type="fa"
      />
       Full screen
    </Button>
  </FormField>
  <Modal
    animation={true}
    autoFocus={true}
    backdrop={true}
    bsClass="modal"
    className="editor-modal"
    dialogComponentClass={[Function]}
    enforceFocus={true}
    keyboard={true}
    manager={
      ModalManager {
        "add": [Function],
        "containers": Array [],
        "data": Array [],
        "handleContainerOverflow": true,
        "hideSiblingNodes": true,
        "isTopModal": [Function],
        "modals": Array [],
        "remove": [Function],
      }
    }
    onHide={[Function]}
    renderBackdrop={[Function]}
    restoreFocus={true}
    show={false}
  >
    <ModalHeader
      bsClass="modal-header"
      closeButton={true}
      closeLabel="Close"
    />
    <ModalBody
      bsClass="modal-body"
      componentClass="div"
    >
      <StringValue
        css="form-control"
        fullScreen={true}
        id="common_parameter_value"
        isMasked={true}
        name="common_parameter[value]"
        onChange={[Function]}
        value=""
      />
    </ModalBody>
  </Modal>
</Fragment>
`;

exports[`ValueField rendering when selectedType = array 1`] = `
<Fragment>
  <FormField
    className=""
    disabled={false}
    error=""
    helpInline={null}
    id={null}
    inputProps={null}
    inputSizeClass="col-md-8"
    label="Value"
    labelHelp={null}
    labelSizeClass="col-md-2"
    onChange={[Function]}
    required={false}
    type="text"
  >
    <Editor
      css="form-control"
      fullScreen={false}
      id="common_parameter_value"
      isMasked={false}
      mode="ruby"
      name="common_parameter[value]"
      onChange={[Function]}
      value=""
    />
    <Button
      active={false}
      block={false}
      bsClass="btn"
      bsStyle="default"
      disabled={false}
      onClick={[Function]}
    >
      <Icon
        name="expand"
        type="fa"
      />
       Full screen
    </Button>
  </FormField>
  <Modal
    animation={true}
    autoFocus={true}
    backdrop={true}
    bsClass="modal"
    className="editor-modal"
    dialogComponentClass={[Function]}
    enforceFocus={true}
    keyboard={true}
    manager={
      ModalManager {
        "add": [Function],
        "containers": Array [],
        "data": Array [],
        "handleContainerOverflow": true,
        "hideSiblingNodes": true,
        "isTopModal": [Function],
        "modals": Array [],
        "remove": [Function],
      }
    }
    onHide={[Function]}
    renderBackdrop={[Function]}
    restoreFocus={true}
    show={false}
  >
    <ModalHeader
      bsClass="modal-header"
      closeButton={true}
      closeLabel="Close"
    />
    <ModalBody
      bsClass="modal-body"
      componentClass="div"
    >
      <Editor
        css="form-control"
        fullScreen={true}
        id="common_parameter_value"
        isMasked={false}
        mode="ruby"
        name="common_parameter[value]"
        onChange={[Function]}
        value=""
      />
    </ModalBody>
  </Modal>
</Fragment>
`;

exports[`ValueField rendering when selectedType = boolean 1`] = `
<Fragment>
  <FormField
    className=""
    disabled={false}
    error=""
    helpInline={null}
    id={null}
    inputProps={null}
    inputSizeClass="col-md-4"
    label="Value"
    labelHelp={null}
    labelSizeClass="col-md-2"
    onChange={[Function]}
    required={false}
    type="text"
  >
    <BooleanValue
      css="form-control"
      id="common_parameter_value"
      isMasked={false}
      name="common_parameter[value]"
      onChange={[Function]}
      value=""
    />
  </FormField>
  <Modal
    animation={true}
    autoFocus={true}
    backdrop={true}
    bsClass="modal"
    className="editor-modal"
    dialogComponentClass={[Function]}
    enforceFocus={true}
    keyboard={true}
    manager={
      ModalManager {
        "add": [Function],
        "containers": Array [],
        "data": Array [],
        "handleContainerOverflow": true,
        "hideSiblingNodes": true,
        "isTopModal": [Function],
        "modals": Array [],
        "remove": [Function],
      }
    }
    onHide={[Function]}
    renderBackdrop={[Function]}
    restoreFocus={true}
    show={false}
  >
    <ModalHeader
      bsClass="modal-header"
      closeButton={true}
      closeLabel="Close"
    />
    <ModalBody
      bsClass="modal-body"
      componentClass="div"
    >
      <BooleanValue
        css="form-control"
        fullScreen={true}
        id="common_parameter_value"
        isMasked={false}
        name="common_parameter[value]"
        onChange={[Function]}
        value=""
      />
    </ModalBody>
  </Modal>
</Fragment>
`;

exports[`ValueField rendering when selectedType = hash 1`] = `
<Fragment>
  <FormField
    className=""
    disabled={false}
    error=""
    helpInline={null}
    id={null}
    inputProps={null}
    inputSizeClass="col-md-8"
    label="Value"
    labelHelp={null}
    labelSizeClass="col-md-2"
    onChange={[Function]}
    required={false}
    type="text"
  >
    <Editor
      css="form-control"
      fullScreen={false}
      id="common_parameter_value"
      isMasked={false}
      mode="ruby"
      name="common_parameter[value]"
      onChange={[Function]}
      value=""
    />
    <Button
      active={false}
      block={false}
      bsClass="btn"
      bsStyle="default"
      disabled={false}
      onClick={[Function]}
    >
      <Icon
        name="expand"
        type="fa"
      />
       Full screen
    </Button>
  </FormField>
  <Modal
    animation={true}
    autoFocus={true}
    backdrop={true}
    bsClass="modal"
    className="editor-modal"
    dialogComponentClass={[Function]}
    enforceFocus={true}
    keyboard={true}
    manager={
      ModalManager {
        "add": [Function],
        "containers": Array [],
        "data": Array [],
        "handleContainerOverflow": true,
        "hideSiblingNodes": true,
        "isTopModal": [Function],
        "modals": Array [],
        "remove": [Function],
      }
    }
    onHide={[Function]}
    renderBackdrop={[Function]}
    restoreFocus={true}
    show={false}
  >
    <ModalHeader
      bsClass="modal-header"
      closeButton={true}
      closeLabel="Close"
    />
    <ModalBody
      bsClass="modal-body"
      componentClass="div"
    >
      <Editor
        css="form-control"
        fullScreen={true}
        id="common_parameter_value"
        isMasked={false}
        mode="ruby"
        name="common_parameter[value]"
        onChange={[Function]}
        value=""
      />
    </ModalBody>
  </Modal>
</Fragment>
`;

exports[`ValueField rendering when selectedType = integer 1`] = `
<Fragment>
  <FormField
    className=""
    disabled={false}
    error=""
    helpInline={null}
    id={null}
    inputProps={null}
    inputSizeClass="col-md-4"
    label="Value"
    labelHelp={null}
    labelSizeClass="col-md-2"
    onChange={[Function]}
    required={false}
    type="text"
  >
    <NumberValue
      css="form-control"
      id="common_parameter_value"
      isMasked={false}
      name="common_parameter[value]"
      onChange={[Function]}
      value=""
    />
  </FormField>
  <Modal
    animation={true}
    autoFocus={true}
    backdrop={true}
    bsClass="modal"
    className="editor-modal"
    dialogComponentClass={[Function]}
    enforceFocus={true}
    keyboard={true}
    manager={
      ModalManager {
        "add": [Function],
        "containers": Array [],
        "data": Array [],
        "handleContainerOverflow": true,
        "hideSiblingNodes": true,
        "isTopModal": [Function],
        "modals": Array [],
        "remove": [Function],
      }
    }
    onHide={[Function]}
    renderBackdrop={[Function]}
    restoreFocus={true}
    show={false}
  >
    <ModalHeader
      bsClass="modal-header"
      closeButton={true}
      closeLabel="Close"
    />
    <ModalBody
      bsClass="modal-body"
      componentClass="div"
    >
      <NumberValue
        css="form-control"
        fullScreen={true}
        id="common_parameter_value"
        isMasked={false}
        name="common_parameter[value]"
        onChange={[Function]}
        value=""
      />
    </ModalBody>
  </Modal>
</Fragment>
`;

exports[`ValueField rendering when selectedType = json 1`] = `
<Fragment>
  <FormField
    className=""
    disabled={false}
    error=""
    helpInline={null}
    id={null}
    inputProps={null}
    inputSizeClass="col-md-8"
    label="Value"
    labelHelp={null}
    labelSizeClass="col-md-2"
    onChange={[Function]}
    required={false}
    type="text"
  >
    <Editor
      css="form-control"
      fullScreen={false}
      id="common_parameter_value"
      isMasked={false}
      mode="ruby"
      name="common_parameter[value]"
      onChange={[Function]}
      value=""
    />
    <Button
      active={false}
      block={false}
      bsClass="btn"
      bsStyle="default"
      disabled={false}
      onClick={[Function]}
    >
      <Icon
        name="expand"
        type="fa"
      />
       Full screen
    </Button>
  </FormField>
  <Modal
    animation={true}
    autoFocus={true}
    backdrop={true}
    bsClass="modal"
    className="editor-modal"
    dialogComponentClass={[Function]}
    enforceFocus={true}
    keyboard={true}
    manager={
      ModalManager {
        "add": [Function],
        "containers": Array [],
        "data": Array [],
        "handleContainerOverflow": true,
        "hideSiblingNodes": true,
        "isTopModal": [Function],
        "modals": Array [],
        "remove": [Function],
      }
    }
    onHide={[Function]}
    renderBackdrop={[Function]}
    restoreFocus={true}
    show={false}
  >
    <ModalHeader
      bsClass="modal-header"
      closeButton={true}
      closeLabel="Close"
    />
    <ModalBody
      bsClass="modal-body"
      componentClass="div"
    >
      <Editor
        css="form-control"
        fullScreen={true}
        id="common_parameter_value"
        isMasked={false}
        mode="ruby"
        name="common_parameter[value]"
        onChange={[Function]}
        value=""
      />
    </ModalBody>
  </Modal>
</Fragment>
`;

exports[`ValueField rendering when selectedType = real 1`] = `
<Fragment>
  <FormField
    className=""
    disabled={false}
    error=""
    helpInline={null}
    id={null}
    inputProps={null}
    inputSizeClass="col-md-4"
    label="Value"
    labelHelp={null}
    labelSizeClass="col-md-2"
    onChange={[Function]}
    required={false}
    type="text"
  >
    <NumberValue
      css="form-control"
      id="common_parameter_value"
      isMasked={false}
      name="common_parameter[value]"
      onChange={[Function]}
      value=""
    />
  </FormField>
  <Modal
    animation={true}
    autoFocus={true}
    backdrop={true}
    bsClass="modal"
    className="editor-modal"
    dialogComponentClass={[Function]}
    enforceFocus={true}
    keyboard={true}
    manager={
      ModalManager {
        "add": [Function],
        "containers": Array [],
        "data": Array [],
        "handleContainerOverflow": true,
        "hideSiblingNodes": true,
        "isTopModal": [Function],
        "modals": Array [],
        "remove": [Function],
      }
    }
    onHide={[Function]}
    renderBackdrop={[Function]}
    restoreFocus={true}
    show={false}
  >
    <ModalHeader
      bsClass="modal-header"
      closeButton={true}
      closeLabel="Close"
    />
    <ModalBody
      bsClass="modal-body"
      componentClass="div"
    >
      <NumberValue
        css="form-control"
        fullScreen={true}
        id="common_parameter_value"
        isMasked={false}
        name="common_parameter[value]"
        onChange={[Function]}
        value=""
      />
    </ModalBody>
  </Modal>
</Fragment>
`;

exports[`ValueField rendering when selectedType = yaml 1`] = `
<Fragment>
  <FormField
    className=""
    disabled={false}
    error=""
    helpInline={null}
    id={null}
    inputProps={null}
    inputSizeClass="col-md-8"
    label="Value"
    labelHelp={null}
    labelSizeClass="col-md-2"
    onChange={[Function]}
    required={false}
    type="text"
  >
    <Editor
      css="form-control"
      fullScreen={false}
      id="common_parameter_value"
      isMasked={false}
      mode="text"
      name="common_parameter[value]"
      onChange={[Function]}
      value=""
    />
    <Button
      active={false}
      block={false}
      bsClass="btn"
      bsStyle="default"
      disabled={false}
      onClick={[Function]}
    >
      <Icon
        name="expand"
        type="fa"
      />
       Full screen
    </Button>
  </FormField>
  <Modal
    animation={true}
    autoFocus={true}
    backdrop={true}
    bsClass="modal"
    className="editor-modal"
    dialogComponentClass={[Function]}
    enforceFocus={true}
    keyboard={true}
    manager={
      ModalManager {
        "add": [Function],
        "containers": Array [],
        "data": Array [],
        "handleContainerOverflow": true,
        "hideSiblingNodes": true,
        "isTopModal": [Function],
        "modals": Array [],
        "remove": [Function],
      }
    }
    onHide={[Function]}
    renderBackdrop={[Function]}
    restoreFocus={true}
    show={false}
  >
    <ModalHeader
      bsClass="modal-header"
      closeButton={true}
      closeLabel="Close"
    />
    <ModalBody
      bsClass="modal-body"
      componentClass="div"
    >
      <Editor
        css="form-control"
        fullScreen={true}
        id="common_parameter_value"
        isMasked={false}
        mode="text"
        name="common_parameter[value]"
        onChange={[Function]}
        value=""
      />
    </ModalBody>
  </Modal>
</Fragment>
`;
